fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulktools/Main.java

```diff
@@ -0,0 +1,278 @@
+                                new ByteArrayInputStream(bufferStream.toByteArray()), bufferStream.size(), partNum++));
+                        bufferStream.reset();
+                    }
+
+                    if (segNum < numberOfFiles - 1 && lineRed == numOfRes4Seg
+                            || !isMore2Read) {
+                        BulkDataUtils.finishMultiPartUpload(cosClient, cosBucketName, segName, uploadId, dataPackTags);
+                        System.out.println("Finished writting for " + segName);
+                        lineRed = 0;
+                        segNum++;
+                        uploadId = null;
+                        partNum = 1;
+                        dataPackTags.clear();
+                    }
+                }
+        }
+
+    }
+
+
+    /**
+     * Main entry
+     * @param args
+     */
+    public static void main(String[] args) {
+        Main m = new Main();
+        try {
+            m.parseArgs(args);
+
+            long start = System.nanoTime();
+            int totalNum = 0;
+
+            AmazonS3 cosClient = BulkDataUtils.getCosClient(cosCredentialIbm, cosApiKey, cosSrvinstId, cosEndpintUrl,
+                    cosLocation);
+            if (cosClient == null) {
+                logger.warning("Failed to get CosClient!");
+                System.exit(1);
+            }
+
+            S3Object item = cosClient.getObject(new GetObjectRequest(cosBucketName, cosFile2Break));
+            try (S3ObjectInputStream s3InStream = item.getObjectContent();
+                    BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream))) {
+                   totalNum = m.getFhirResourceNumberFromBufferReader(resReader);
+               } catch (Exception ioe) {
+                   logger.warning("Error proccesing file " + cosFile2Break + " - " + ioe.getMessage());
+                   System.exit(1);
+               }
+
+            ByteArrayOutputStream bufferStream = new ByteArrayOutputStream();
+            int numOfRes4Seg = totalNum/numberOfFiles;
+
+            item = cosClient.getObject(new GetObjectRequest(cosBucketName, cosFile2Break));
+            try (S3ObjectInputStream s3InStream = item.getObjectContent();
+                    BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream))) {
+                m.writeFhirResourceFromBufferReader(resReader, cosClient, bufferStream, numOfRes4Seg);
+               } catch (Exception ioe) {
+                   logger.warning("Error proccesing file " + cosFile2Break + " - " + ioe.getMessage());
+                   System.exit(1);
+               }
+
+            long end = System.nanoTime();
+            logger.info(String.format("Total Resources: %d, Took: %6.3f seconds", totalNum, (end-start)/NANOS));
+        }
+        catch (Exception x) {
+            logger.log(Level.SEVERE, "Failed to run", x);
+        }
+        finally {
+            System.exit(0);
```

Comment:
I don't think this is necessary... Let it naturally fall out.

Reply:
I added this on purpose for running inside eclipse, without this, in my tests, eclipse always shows the app is not finished - very annoying.